File: src/AudioraMobile/app/src/main/java/com/audiora/app/models/PagedResponse.java
package com.audiora.app.models;

import java.util.List;

public class PagedResponse<T> {
    private List<T> data;
    private int page;
    private int pageSize;
    private int totalCount;
    private int totalPages;
    private boolean hasNext;
    private boolean hasPrevious;

    public List<T> getData()    { return data; }
    public int getTotalCount()  { return totalCount; }
    public int getTotalPages()  { return totalPages; }
    public boolean isHasNext()  { return hasNext; }
}

File: src/Audiora.Admin/Models/PagedResponse.cs
namespace Audiora.Admin.Models;

public class PagedResponse<T>
{
    public List<T> Data        { get; set; } = new();
    public int     TotalCount  { get; set; }
    public int     TotalPages  { get; set; }
    public bool    HasNext     { get; set; }
}